jina/peapods/peas/jinad.py

```diff
@@ -249,6 +286,7 @@ def _terminate(self):
         This method calls :meth:`terminate` in :class:`threading.Thread` or :class:`multiprocesssing.Process`.
         """
         self.cancel_event.set()  # Inform JinaD Process to stop streaming
+        self.is_shutdown.wait()  # Wait until JinaD terminates remote Pea and sets shutdown event
```

Comment:
No, this should not be here, u should terminate the `worker` and handle it properly there